crates/services/txpool_v2/src/service.rs

```diff
@@ -463,6 +486,25 @@ where
                     Error::Removed(RemovedReason::LessWorth(tx.id())),
                 );
             }
+        };
+        move || {
+            if metrics {
+                let txpool_metrics = txpool_metrics();
+                txpool_metrics
+                    .number_of_transactions_pending_verification
+                    .inc();
```

Comment:
You will increase the number of pending transactions in the queue only when you start to execute this function that doesn't make a lot of sense. We need to increase number of pending transactions before you insert transaction.

Reply:
Reworked in https://github.com/FuelLabs/fuel-core/pull/2385/commits/959c4b9625dee9c6e92a179281d7c74d7d6dfc14

@AurelienFT, have a look at this one, please :pray: